packages/web-config-server/src/aggregator/buildAggregationOptions.js

```diff
@@ -84,3 +86,32 @@ const fetchEntityAggregationConfig = async (
     config: { ...entityAggregationConfig, orgUnitMap: entityToAncestorMap },
   };
 };
+
+const replaceEntityAnswerWithEntityNameConfig = async (
+  models,
+  dataSourceEntities,
```

Comment:
I don't think this will work in general... `dataSourceEntities` are built based on the entities selected in the `web-config-server`, in other words the entity selected in the front end.

On the other hand, your code concerns `QMIA029`, which is an `Entity` question. In the general case, this entity could be anywhere in the world! For all we know, a Samoa survey may allow the user to select a city in Australia.

In conclusion, this **will** work but only under specific conditions. What would be more useful instead would be to map any entity id to its name, since we already have this information in the database.